src/main/java/slimeknights/tconstruct/tools/common/item/ItemBlockTable.java

```diff
@@ -34,7 +39,27 @@ public void addInformation(@Nonnull ItemStack stack, @Nonnull EntityPlayer playe
     }
 
     if(stack.getTagCompound().hasKey("inventory")) {
-      tooltip.addAll(LocUtils.getTooltips(Util.translate("tooltip.chest.has_items")));
+      // get the inventory, so we can inspect its items
+      NBTTagCompound inventory = stack.getTagCompound().getCompoundTag("inventory");
+
+      if (!inventory.hasKey("Items")) {
+        return;
+      }
+
+      NBTTagList items = inventory.getTagList("Items", 10);
+
+      // check if it's a PatternChest (damage value corresponds to TableTypes enum)
+      if (BlockToolTable.TableTypes.fromMeta(stack.getItemDamage()) == BlockToolTable.TableTypes.PatternChest) {
+        // determine if it holds casts or patterns
+        // assume the contents are valid, thus checking any stack in the list suffices
+        Item item = ItemStack.loadItemStackFromNBT(items.getCompoundTagAt(0)).getItem();
```

Comment:
in which case we'd need the loop again...

Say: slot 0 -> removed item, slot 1 -> valid cast/pattern